app/models/sms_message.rb

```diff
@@ -52,4 +52,70 @@ class SmsMessage < ApplicationRecord
                   received: "received",
                   read: "read",
   }, prefix: :status
+
+  def self.save_api_response(message, recipient)
+    retries ||= 0
+    record = find_or_initialize_by(message_sid: message.sid)
+
+    create_only_fields = {
+      body: message.body,
+      recipient: recipient
+    }
+
+    create_or_update_fields = {
+      from: message.from,
+      to: message.to,
+      date_created: message.date_created,
+      date_updated: message.date_updated,
+      date_sent: message.date_sent,
+      direction: message.direction,
```

Comment:
Why is `direction` not in  `create_only_fields`?

Reply:
Good point - it can't change in an status update and is set immediately, so it should be there. I'll move it.

Reply:
(If you want to relay this comment to the GCF or GYR teams btw, might be useful for them although I don't think this will ever result in incorrect behavior _unless_ you wanted to check direction on a scheduled message that hadn't received a status update yet)